### PR TITLE
map is hidden behind a button now, photos and comments are only shown…

### PR DESCRIPTION
**Pull request by RoniNiklas**
map is hidden by default
get directions button opens a modal with the map
photos and comments in the restaurantEntry are only shown for restaurants with placeid
RouteMap now takes a new number prop called size, which defines the map's height and width.

### CODE REVIEW

File: packages/app/src/components/MapModal/MapModal.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <Modal data-testid='map-modal' show={showMap} onHide={() => setShowMap(false)} animation={false}>
+      <Modal.Header data-testid='modal-header' closeButton>
+                Directions to {restaurant.name}
+      </Modal.Header>
+      <Modal.Body>
+        {restaurant.coordinates
+          ? <RouteMap restaurant={restaurant} size={550} />
+          :  <div data-testid="mapmodal-error"> That restaurant does not have an address or coordinates in the database! </div>
```
Comment (ritinae) [inline packages/app/src/components/MapModal/MapModal.js]: This  error message is a bit technical from user point of view

File: packages/app/src/components/RouteMap/RouteMap.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ const RouteMap = ({ restaurant }) => {
         tileSize={512}
         zoomOffset={-1}
         zoom={14}
+        style={{width:size+'px', height:size+'px'}}
```
Comment (ritinae) [inline packages/app/src/components/RouteMap/RouteMap.js]: I think this should be in CSS? Inline styles are a bit iffy. Selectors could be used then to control the size in different contexts.

**Review by ritinae — DISMISSED** [review]

File: packages/app/src/components/MapModal/MapModal.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const MapModal = ({ restaurant, showMap, setShowMap }) => {
+
+  return (
+    <Modal data-testid='map-modal' show={showMap} onHide={() => setShowMap(false)} animation={false}>
+      <Modal.Header data-testid='modal-header' closeButton>
+                Directions to {restaurant.name}
+      </Modal.Header>
+      <Modal.Body>
+        {restaurant.coordinates
+          ? <RouteMap restaurant={restaurant} size={550} />
```
Comment (otsha) [inline packages/app/src/components/MapModal/MapModal.js]: Where does this magic number come from?

File: packages/app/src/components/RestaurantEntry/RestaurantEntry.css [hunk 4]
```diff
@@ -45,4 +45,16 @@
...
+  margin: 0.5em;
+  background-color: Transparent;
+  outline:none;
+  color:rgba(255,255,255,0.9); 
+  border:none;
+}
+.restaurantentry-showmap-button:hover {
+  text-decoration: underline;
+  color:rgba(255,255,255,1); 
+}
```
Comment (otsha) [inline packages/app/src/components/RestaurantEntry/RestaurantEntry.css:60]: Bootstrap also has a link variant for buttons, but this works just as well.

File: packages/app/src/components/Restaurants/form/RestaurantForm.js [hunk 5]
```diff
@@ -200,7 +200,7 @@ const RestaurantForm = ({ restaurant, setRestaurant, onSubmit, submitMessage = '
             {errors.address && renderError(errors.address)}
           </Form.Group>
 
-          {restaurant.coordinates && <RouteMap restaurant={restaurant} />}
+          {restaurant.coordinates && <RouteMap restaurant={restaurant} size={420} />}
```
Comment (otsha) [inline packages/app/src/components/Restaurants/form/RestaurantForm.js]: Magic number

File: packages/app/src/components/RouteMap/RouteMap.test.js [hunk 6]
```diff
@@ -22,7 +22,7 @@ const testRestaurant = {
 
 test('map component renders without crashing when provided with good props', async () => {
   const { getByTestId } = await actRender(
-    <RouteMap restaurant={testRestaurant} />
+    <RouteMap restaurant={testRestaurant} size={100} />
```
Comment (otsha) [inline packages/app/src/components/RouteMap/RouteMap.test.js]: Magic number?

**Review by otsha — COMMENTED** [review]
Modal looks great on both mobile & desktop.

A few issues with "magic numbers" along with what Ilari commented.

**Review by otsha — APPROVED** [review]

**Pull request merged**